### PR TITLE
74 update react to v18

### PR DESCRIPTION
**Pull request by Elkrival**
Update React version to 18 and updated rendering method.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,7 @@ server/configs
 docs/build
 public/js
 dist
+
+#Vagrant
```
Comment (mikestone14) [inline .gitignore]: what brought about these changes to the .gitignore?

File: views/Account.render.react.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from 'react'
-import { render } from 'react-dom'
+import { createRoot } from 'react-dom/client';
```
Comment (mikestone14) [inline views/Account.render.react.js]:
```suggestion
import { createRoot } from 'react-dom/client'
```

File: views/Account.render.react.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 import { Provider } from 'react-redux'
-
 import AccountPage from './Account.react'
 import store from '../stores/store'
 
 const reduxStore = store()
+const container = document.getElementById('main');
+const root = createRoot(container);
```
Comment (mikestone14) [inline views/Account.render.react.js]:
```suggestion
const container = document.getElementById('main')
const root = createRoot(container)
```

File: views/Account.render.react.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
 
-render(<Provider store={reduxStore}><AccountPage /></Provider>, document.getElementById('main'));
+root.render(<Provider store={reduxStore}><AccountPage /></Provider>);
```
Comment (mikestone14) [inline views/Account.render.react.js]:
```suggestion
root.render(<Provider store={reduxStore}><AccountPage /></Provider>)
```

File: views/Main.render.react.js [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
 import MainPage from './Main.react';
 import store from '../stores/store';
 
+const container = document.getElementById('main');
+const root = createRoot(container);
 const reduxStore = store();
 
-render(<Provider store={reduxStore}><MainPage /></Provider>, document.getElementById('main'));
+root.render(<Provider store={reduxStore}><MainPage /></Provider>);
+
```
Comment (mikestone14) [inline views/Main.render.react.js:10]:
```suggestion
```

**Review by mikestone14 — COMMENTED** [review]
This looks good I think there is just some formatting cleanup, mostly removing the semi-colons that were added.

**Comment by mikestone14** [comment]
Also want to confirm that you were able to test this change in the Centos/Singularity environment

File: .gitignore (hunk 1 above)
Comment (Elkrival) [reply .gitignore]: This was removed, found a better way to test the clients environment. 
[8b188e4](https://github.com/AMP-SCZ/dpdash/pull/75/commits/8b188e4dd3522e4ca13f10b2083b5a0224dfaabf)

File: views/Account.render.react.js (hunk 2 above)
Comment (Elkrival) [reply views/Account.render.react.js]: [8b188e4](https://github.com/AMP-SCZ/dpdash/pull/75/commits/8b188e4dd3522e4ca13f10b2083b5a0224dfaabf)

File: views/Account.render.react.js (hunk 3 above)
Comment (Elkrival) [reply views/Account.render.react.js]: [8b188e4](https://github.com/AMP-SCZ/dpdash/pull/75/commits/8b188e4dd3522e4ca13f10b2083b5a0224dfaabf)

File: views/Account.render.react.js (hunk 4 above)
Comment (Elkrival) [reply views/Account.render.react.js]: [8b188e4](https://github.com/AMP-SCZ/dpdash/pull/75/commits/8b188e4dd3522e4ca13f10b2083b5a0224dfaabf)

File: views/Main.render.react.js (hunk 5 above)
Comment (Elkrival) [reply views/Main.render.react.js:10]: [8b188e4](https://github.com/AMP-SCZ/dpdash/pull/75/commits/8b188e4dd3522e4ca13f10b2083b5a0224dfaabf)

**Comment by Elkrival** [comment]
> Also want to confirm that you were able to test this change in the Centos/Singularity environment

@mikestone14  Yes this has been tested in an environment that replicates the clients.

File: package.json [hunk 6]
```diff
@@ -88,11 +89,14 @@
     "@babel/register": "^7.13.16",
     "babel-loader": "^8.2.1",
     "babel-plugin-react-html-attrs": "^3.0.5",
+    "connect-livereload": "^0.6.1",
```
Comment (mikestone14) [inline package.json]: weren't these added in a separate PR?

Comment (Elkrival) [reply package.json]: [a15fa1a](https://github.com/AMP-SCZ/dpdash/pull/75/commits/a15fa1abd5022937b71b31ee0b92525bad9f9ab0)

**Review by mikestone14 — APPROVED** [review]

**Pull request merged**